### ISSUE TITLE
Status? Objective? Progress? Crash?

### ISSUE THREAD

**Issue by Zomis** [Anders142536/Foogle#3]
I found the name of this mod interesting, but I notice nothing has happened with it for the last 1+ years.

So I am wondering: What is the status of this mod? Is it abandoned? Also, what is the objective for this mod? I don't see any readme so it's hard to tell what the purpose of it is. From the mod-info it looks like it has a very specific purpose, only finding out where iron plates leak to the logistic network?

Also, there's a crash when trying it with Factorio 1.1.74:

```
The mod Foogle (0.0.1) caused a non-recoverable error.
Please report this error to the mod author.

Error while running event Foogle::on_init()
__Foogle__/control.lua:14: attempt to concatenate local 'player' (a table value)
stack traceback:
	__Foogle__/control.lua:14: in function 'initializePlayer'
	__Foogle__/control.lua:25: in function 'initialize'
	__Foogle__/control.lua:36: in function <__Foogle__/control.lua:34>
```

**Comment by tburrows13** [comment]
I believe the intent was/is to make 'factorio google'. See https://mods.factorio.com/mod/BeastFinder/discussion/60101eb93bc6e52c26aced49 and https://mods.factorio.com/mod/FactorySearch/discussion/624abf61c0aea5d9c45ee98d :)

I hope the house has gone well @Anders142536!

**Comment by Anders142536** [comment]
Yes, you are right, it was meant to be a Factorio Google. I didn't add a proper description as I didn't know the full scope of the mod yet.

I wanted to do this as I was super unhappy with the BeastFinder UI and its author not minding me using their code for searching. Shortly after I wanted to do this [Factory Search](https://mods.factorio.com/mod/FactorySearch) was released with good activity and solving many of the issues I had with the UI of BeastFinder. I might start working on this again, I don't know yet. I haven't decided to abandon this idea yet.

The house - which is the main source of stress preventing me from doing work on my mods - is still not built. The finance market in europe is catastrophic for the project. I look forward to come back to modding more actively on the weekends, but I will not for at least another year, I suppose.

**Comment by Zomis** [comment]
Ok, I understand, and suspected it would do something like that :)

I was planning on making my own search mod as well, as I was also unhappy with Beast Finder. I started this: https://github.com/Zomis/FactorioMods/tree/main/search-engine_0.0.1 but it's a horrible GUI but has several features, and I think the code architecture of it is quite expandable.

My intentions was to make a more advanced search where you could have various filters and compute e.g. sum of items, or aggregate the results in other ways. But that has stalled for now.

FYI, I've created another mod with a similar name as Foogle, but the intention of it is not to be a search engine but rather be a link between different mods. https://mods.factorio.com/mod/foofle (there's still some UI work to do there and more integrations to add - currently it only supports FNEI and Train Log).

Best of luck with the house! I'm happy to discuss my wishes for a search mod if you're interested.

**Comment by Anders142536** [comment]
It's good to see competition rise up. My idea was to create a "search library" based on BeastFinders (and maybe Factory Search's or your) code and reduce BeastFinder itself to a frontend for it. Foogle would be a frontend for the search library as well. Once I have more time I would love to do that. Would you be interested in using something like that for your mod as well?

Foofle sounds like an interesting approach. Maybe I will add support for it in my mods.

**Comment by Zomis** [comment]
I would be more interested in making the search library than making the frontend for it, GUIs are not my best subject... 😅

**Comment by Anders142536** [comment]
I'm open to other helping once I start with it. I'm a frontend dev by trade, so I prefer the GUI stuff. That would be a nice fit.

Unless I forget I will notify you once I start working on the lib so you can help out if you would like to.
